Tools/TestWebKitAPI/Tests/WebKitCocoa/CreateWebArchive.mm

```diff
@@ -949,6 +949,139 @@ function frameLoaded() {
     Util::run(&saved);
 }
 
+static const char* htmlDataBytesForShadowDOM = R"TESTRESOURCE(
+<button id="button" onclick="buttonClicked()"></button>
+<script>
+function buttonClicked() {
+    div1 = document.createElement("div");
+    div1.id = "div1";
+    root1 = div1.attachShadow({ mode: "open" });
+    root1.innerHTML = "ShadowRoot1";
+    div2 = document.createElement("div");
+    div2.id = "div2";
+    root2 = div2.attachShadow({ mode: "closed" });
+    root2.innerHTML = "ShadowRoot2";
+    root1.appendChild(div2);
```

Comment:
Can we add a slot element to make sure it gets serialized properly?

Reply:
Sure, will add.